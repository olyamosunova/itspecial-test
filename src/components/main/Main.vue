<template>
    <section class="container">
        <FormFilter />

        <Loader v-if="!data" />
        <Table v-else />

        <UserInfo v-if="chosenUser" />
    </section>
</template>

<script>
    import FormFilter from "../form-filter/Form-filter";
    import Table from "../table/Table";
    import Loader from "../loader/Loader";
    import UserInfo from "../user-info/User-info";

    export default {
        name: "Main",
        mounted() {
            this.$store.dispatch('getData');
        },
        computed: {
            data() {
                return this.$store.getters.data;
            },
            chosenUser() {
                return this.$store.getters.chosenUser;
            }
        },
        components: {
            Loader,
            FormFilter,
            Table,
            UserInfo
        },
    }
</script>

<style scoped>

</style>
