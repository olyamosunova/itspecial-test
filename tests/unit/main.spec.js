import { mount, createLocalVue } from "@vue/test-utils";
import Main from "@/components/main/Main";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

const data = [
    {
        "id": 261,
        "firstName": "Meribeth",
        "lastName": "Haglund",
        "email": "VDaly@odio.io",
        "phone": "(411)804-5726",
        "adress": {
            "streetAddress": "9749 Ante Ct",
            "city": "El Dorado Hills",
            "state": "NV",
            "zip": "71328"
        },
        "description": "aliquam lacus odio magna lacus sed sed lacus sed convallis ipsum sollicitudin libero magna hendrerit tortor magna porta orci sed aliquam sed vitae magna egestas placerat curabitur amet suspendisse lacus sagittis sollicitudin"
    },
    {
        "id": 295,
        "firstName": "Marcus",
        "lastName": "Espinosa",
        "email": "LDover@mattis.com",
        "phone": "(497)561-3057",
        "adress": {
            "streetAddress": "5763 Lorem Ln",
            "city": "Walnut Creek",
            "state": "TX",
            "zip": "92471"
        },
        "description": "pharetra sed sit eros augue nec risus porttitor morbi sit sit egestas sit sed dolor pretium vitae magna amet sed rutrum vitae mi amet convallis vitae consectetur massa tincidunt quis sapien vel"
    },
    {
        "id": 289,
        "firstName": "Ravi",
        "lastName": "Kramarsky",
        "email": "AGerver@sed.ly",
        "phone": "(484)242-1514",
        "adress": {
            "streetAddress": "8851 Placerat St",
            "city": "Oakland",
            "state": "UT",
            "zip": "80078"
        },
        "description": "mattis vel sed ante dui nunc in placerat ipsum vitae nunc morbi sed adipiscing ac vel sollicitudin orci pulvinar sed amet dolor massa vitae turpis sit ac et massa augue tortor amet"
    }
];

describe('unit tests for Main component', () => {
    let actions;
    let getters;
    let store;

    beforeEach(() => {
        actions = {
            getData: jest.fn(),
        };

        getters = {
            data: () => data,
            chosenUser: () => data[0]
        };

        store = new Vuex.Store({
            actions,
            getters
        });
    });

    it('should match the snapshot', () => {
        const wrapper = mount(Main, { store, localVue });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
