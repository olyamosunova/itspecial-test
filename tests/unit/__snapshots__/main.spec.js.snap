// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for Main component should match the snapshot 1`] = `
<section class="container">
  <form>
    <div class="input-group mb-3"><input type="text" placeholder="Enter text" aria-label="Enter text" required="required" class="form-control"> <button type="submit" class="btn btn-primary">Find</button></div>
  </form>
  <div>
    <div class="overflow-auto mt-5 mb-5">
      <table class="table table-hover">
        <thead class="thead-light">
          <tr>
            <th scope="col" class="sort">
              ID
              <span class="sort-icon"></span>
            </th>
            <th scope="col" class="sort">
              First Name
              <span class="sort-icon"></span>
            </th>
            <th scope="col" class="sort">
              Last Name
              <span class="sort-icon"></span>
            </th>
            <th scope="col" class="">
              Phone
              <!---->
            </th>
            <th scope="col" class="">
              Email
              <!---->
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <!---->
          </tr>
        </tbody>
      </table>
    </div>
    <!---->
  </div>
  <div class="card mt-5">
    <div class="card-body">
      <p class="card-text">Selected user: <b>Meribeth Haglund</b></p>
      <p class="card-text">Description:
        <textarea disabled="disabled" class="form-control"></textarea>
      </p>
      <p class="card-text">Residence address: <b>9749 Ante Ct</b></p>
      <p class="card-text">City: <b>El Dorado Hills</b></p>
      <p class="card-text">Province / State: <b>NV</b></p>
      <p class="card-text">Zip: <b>71328</b></p>
    </div>
  </div>
</section>
`;
